src/iter.rs

```diff
@@ -93,8 +94,12 @@ impl<'a> io::Read for HexToBytesIter<'a> {
 
 /// `hi` and `lo` are bytes representing hex characters.
 fn hex_chars_to_byte(hi: u8, lo: u8) -> Result<u8, HexToBytesError> {
-    let hih = (hi as char).to_digit(16).ok_or(HexToBytesError::InvalidChar(hi))?;
-    let loh = (lo as char).to_digit(16).ok_or(HexToBytesError::InvalidChar(lo))?;
+    let hih = (hi as char)
+        .to_digit(16)
+        .ok_or(HexToBytesError::InvalidChar(InvalidCharError { invalid: hi }))?;
```

Comment:
You can keep it short by writing `ok_or(InvalidCharError { invalid: hi })?;` thanks to the conversion made by `?`.

Reply:
I back and forth'ed on this a bunch of times (literally added and remove code). It works sometimes and in some places it didn't so I went with uniform and wrote it out fully everywhere. I"ll have another go though.

Reply:
Oh I misspoke, you suggestion is clearly better. I was referring to the `into` case below.